File: module/module-sqlanalysis/src/main/java/org/sunt/sqlanalysis/model/EmptyTable.java
package org.sunt.sqlanalysis.model;

import java.util.Collections;
import java.util.List;

public class EmptyTable implements FromItem{

    private String expression = "INVALID_TABLE";

    public EmptyTable(String expression) {
        this.expression = expression;
    }

    public EmptyTable() {
    }

    @Override
    public List<SelectItem> getFields() {
        return Collections.emptyList();
    }

    @Override
    public String getLabel() {
        return "";
    }

    @Override
    public List<Table> getSourceTable() {
        return Collections.emptyList();
    }

    @Override
    public String toString() {
        return expression;
    }
}

File: module/module-sqlanalysis/src/main/java/org/sunt/sqlanalysis/lineage/model/table/EmptyTable.kt
package org.sunt.sqlanalysis.lineage.model.table

import org.sunt.sqlanalysis.lineage.model.field.IField

class EmptyTable(val name: String) : LogicalTable(emptyList()) {

    override var alias: String = name

    override val fields: List<IField> = emptyList()

}